frontend/ui/widget/container/inputcontainer.lua

```diff
@@ -269,6 +269,64 @@ function InputContainer:onGesture(ev)
     end
 end
 
+-- Will be overloaded by the Gestures plugin, if enabled, for use in _onGestureFiltered
+function InputContainer:isGestureAlwaysActive(ges, multiswipe_directions)
+    -- If the plugin isn't enabled, block everything, we'll restore standard behavior on resume.
+    return false
+end
+
+-- Filtered variant that only lets is_always_active touch zones through.
+-- (This is used by the "toggle_touch_input" Dispatcher action).
+function InputContainer:_onGestureFiltered(ev)
+    for _, tzone in ipairs(self._ordered_touch_zones) do
+        if self:isGestureAlwaysActive(tzone.def.id, ev.multiswipe_directions) and tzone.gs_range:match(ev) and tzone.handler(ev) then
+            return true
+        end
+    end
+    -- No ges_events at all, although if the need ever arises, we could also support an is_always_active field for those ;).
+    if self.stop_events_propagation then
+        return true
+    end
+end
+
+-- NOTE: This works because barely anything overloads onGesture.
+--       We would need to be slightly more creative if things did
+--       (i.e., don't stop the event and swap self.onGesture in every instance
+--       while still only swapping Input.onGesture once...)
+function InputContainer:onIgnoreTouchInput(toggle)
+    local Notification = require("ui/widget/notification")
+    if toggle == false then
+        -- Restore the proper onGesture handler if we disabled it
+        if InputContainer._onGesture then
+            InputContainer.onGesture = InputContainer._onGesture
+            InputContainer._onGesture = nil
+            Notification:notify("Restored touch input")
+        end
+    elseif toggle == true then
+        -- Replace the onGesture handler w/ the minimal one if that's not already the case
+        if not InputContainer._onGesture then
+            InputContainer._onGesture = InputContainer.onGesture
+            InputContainer.onGesture = InputContainer._onGestureFiltered
```

Comment:
I see we have (that you added probably) a lot of:
```
frontend/apps/reader/modules/readertoc.lua:function ReaderToc:onGesture() end
frontend/apps/reader/modules/readerconfig.lua:function ReaderConfig:onGesture() end
frontend/apps/reader/modules/readermenu.lua:function ReaderMenu:onGesture() end
frontend/apps/reader/modules/readerlink.lua:function ReaderLink:onGesture() end
frontend/apps/reader/modules/readerrotation.lua:function ReaderRotation:onGesture() end
frontend/apps/reader/modules/readerhighlight.lua:function ReaderHighlight:onGesture() end
frontend/apps/reader/modules/readerbookmark.lua:function ReaderBookmark:onGesture() end
frontend/apps/reader/modules/readerfont.lua:function ReaderFont:onGesture() end
frontend/apps/reader/modules/readerpaging.lua:function ReaderPaging:onGesture() end
frontend/apps/reader/modules/readerrolling.lua:function ReaderRolling:onGesture() end
```
How do these relate to what you are doing (you won't override these no-op which take precedence over your superclass monkey patch) ?

Reply:
They aren't strictly related, hence the separate commit (or, actually, PR).

This is related to what #9710 unearthed, and was cleared up in 1e24a1c7a32f53329962ba503c969fcf690b800a

Reply:
Basically, it's for ReaderUI modules that don't actually register their own gestures, but defer everything to ReaderUI.

Instead of having sendEvent go through a couple of empty arrays in the default InputContainer implementation, it just goes through a NOP implementation for those.

Reply:
> it just goes through a NOP implementation for those

So, as it's a no-op, one that does not need to get overridden and replaced by your filtered one, right?

Reply:
The very fact that it's implemented in a subclass means that we don't have anything extra to do (which is similar to what the comment is trying to convey), as we're only replacing the method of the base InputContainer class (which is also why we only do it once).

If we actually had meaningful ongesture subclass implementations[1], we'd need a broadcast event, and to replace self.onGesture, too.

[1] Notifications has one, but I figured dismissing notifs on tap wasn't really a hindrance ;).